### PR TITLE
refactor(Scripts/SSC): Lady Vashj boss script updated

### PR DESCRIPTION
**Pull request by elthehablo**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

repeat maxtimer needs work. average is skewed because the max is very high. I took the average timer as maxtimer now

## Issues Addressed:

N.A.

## SOURCE:

The changes have been validated through:
N.A.

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [x] This pull request requires further testing and may have edge cases to be tested.

## Known Issues and TODO List:

- [x] sniffs

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp [hunk 1]
```diff
@@ -88,215 +74,224 @@ class startFollow : public BasicEvent
...
 
-        void KilledUnit(Unit* /*victim*/) override
-        {
-            if (events.GetNextEventTime(EVENT_KILL_TALK) == 0)
-            {
-                Talk(SAY_SLAY);
-                events.ScheduleEvent(EVENT_KILL_TALK, 6000);
-            }
-        }
+        me->CastSpell(me, SPELL_REMOVE_TAINTED_CORES, true);
```
Comment (Nefertumm) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp:139]: DoCastSelf

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp [hunk 2]
```diff
@@ -88,215 +74,224 @@ class startFollow : public BasicEvent
...
+    }
+
+    void JustSummoned(Creature* summon) override
+    {
+        summons.Summon(summon);
+        if (summon->GetEntry() == WORLD_TRIGGER)
         {
-            Talk(SAY_DEATH);
-            BossAI::JustDied(killer);
+            summon->CastSpell(summon, SPELL_MAGIC_BARRIER, false);
```
Comment (Nefertumm) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp:149]: triggered is false by default

**Review by Nefertumm — APPROVED** [review]
Minor changes, also what about the schedules on MovementInform? Is the boss already in combat? If not, you should move the scheduler update before the updatevictim

**Review by Nefertumm — APPROVED** [review]

**Comment by elthehablo** [comment]
> Minor changes, also what about the schedules on MovementInform? Is the boss already in combat? If not, you should move the scheduler update before the updatevictim

Yep the boss is already in combat way before that. It's the second phase

**Pull request merged**